web_ui/src/shared/components/media-preview-list/media-preview-list.component.tsx

```diff
@@ -1,95 +1,77 @@
     items,
     height,
     viewMode,
-    hasItemPreview = false,
+    selectedItems = {},
     hasLabelSelector = true,
-    onDeleteItem,
+    onPress,
     onUpdateItem,
+    topLeftElement,
+    topRightElement,
 }: MediaPreviewListProps<T>): JSX.Element => {
-    const dialogState = useOverlayTriggerState({});
-    const [previewIndex, setPreviewIndex] = useState<null | number>(0);
-
     return (
-        <>
-            <MediaItemsList
-                viewMode={viewMode}
-                mediaItems={items}
-                height={height}
-                idFormatter={(item) => item.id}
-                getTextValue={(item) => item.file.name}
-                itemContent={(item) => {
-                    const { id, dataUrl, labelIds, file } = item;
+        <MediaItemsList
+            viewMode={viewMode}
+            mediaItems={items}
+            height={height}
+            idFormatter={getId}
+            getTextValue={(item) => item.file.name}
+            itemContent={(item) => {
+                const { id, dataUrl, labelIds, file } = item;
 
-                    return (
-                        <MediaItem
-                            id={id}
-                            key={id}
-                            onPress={() => {
-                                setPreviewIndex(items.findIndex(hasEqualId(id)));
-                                dialogState.open();
-                            }}
-                            height={'100%'}
-                            url={dataUrl}
-                            mediaFile={file}
-                            labelIds={labelIds}
-                            viewMode={viewMode}
-                            hasItemPreview={hasItemPreview}
-                            hasLabelSelector={hasLabelSelector}
-                            onDeleteItem={onDeleteItem}
-                            onSelectLabel={(newLabels) => {
-                                if (isEmpty(newLabels)) {
-                                    onUpdateItem(id, { ...item, labelIds: [], labelName: null });
-                                } else {
-                                    const newLabelIds = getIds(newLabels);
-                                    const newLabelName = newLabels.at(-1)?.name || null;
+                return (
+                    <MediaItem
+                        id={id}
+                        key={id}
+                        onPress={onPress}
+                        height={'100%'}
+                        url={dataUrl}
+                        mediaFile={file}
+                        labelIds={labelIds}
+                        viewMode={viewMode}
+                        isSelected={selectedItems[id] ?? false}
+                        hasLabelSelector={hasLabelSelector}
+                        topLeftElement={topLeftElement}
+                        topRightElement={topRightElement}
+                        onSelectLabel={(newLabels) => {
```

Comment:
are we gonna apply the same logic to the label selector as well? I.e. use renderProps